docs/components/registrations.md

```diff
@@ -0,0 +1,138 @@
+        }
+        // ..
+    }
+    // ..
+}
+```
+
+It's good practice to ignore zero values, as that allows providing components to skip
+the registration if desired.
+
+Finally, the providing component (in this case, `foo`) includes a registration in its output as an additional provided type, beyond its `Component` type:
+
+```go
+// --- foo/foo.go ---
+func newFoo(deps dependencies) (Component, server.Registration) {
+    // ..
+    return foo, server.NewRegistration("/things/foo", foo.handler)
+}
+```
+
+This technique has some caveats to be aware of:
+
+ * The providing components are instantiated before the collecting component.
+ * Fx treats value groups as the collecting component depending on all of the providing components.
+   This means that the providing components cannot depend on the collecting component, as this would represent a dependency cycle.
+ * Fx will instantiate _all_ declared providing components before the collecting component, regardless of whether their `Component` type is required.
+   This may lead to components being instantiated in unexpected circumstances.
+
+## Subscriptions
+
+Subscriptions are a common form of registration, and have support in the `pkg/util/subscriptions` package.
+
+To implement a subscription, the collecting component defines a message type.
+Providing components provide `subscriptions.Subscription[coll.Message]`, from which they can obtain a `subscriptions.Receiver[coll.Message]`.
+Collecting components require `subcriptions.Publisher[coll.Message]`, from which they can obtain a `subscriptions.Transmitter[coll.Message]`.
+
+```go
+// --- announcer/component.go ---
+
+// ...
+// To subscribe to these announcements, provide a subscriptions.Subscription[announcer.Announcement].
+// ...
+package announcer
+```
+
+```go
+// --- announcer/announcer.go ---
+
+func newAnnouncer(pub subscriptions.Publisher[Anouncement]) Component {
+    return &announcer{announcementTx: pub.Transmitter()}  // (get a Transmitter from the Publisher)
+}
+
+// .. later send messages with 
+func (ann *announcer) announce(a announcement) {
+    ann.eventTx.Notify(a)
+}
+```
+
+```go
+// --- listener/listener.go ---
+
+func newListener() (Component, subscriptions.Subscription[announcer.Announcement]) {
+    sub := subscriptions.NewSubscription[Event]() // create a subscription
+    return &listener{eventRx: sub.Receiver}, sub  // capture the receiver, and return the subscription
+}
+
+// .. later receive messages (usually in an actor's main loop)
+func (l *listener) run() {
+    loop {
+        select {
+        case a := <- l.eventRx.Chan():
+            ...
+        }
+    }
+}
+```
+
+If a receiving component decides it does not want to subscribe after all (such as, if it is not started), it can return the zero value, `subscriptions.Subscription[Event]{}`, from its constructor.
+If a component returns a non-nil subscriber, though, then it _must_ consume messages from the receiver or risk blocking the transmitter.
```

Comment:
```suggestion
If a receiving component does not subscribe (for example, if it is not started), it can return the zero value, `subscriptions.Subscription[Event]{}`, from its constructor.
If a component returns a non-nil subscriber, it _must_ consume messages from the receiver or risk blocking the transmitter.
```